packages/@aws-cdk/aws-lambda-nodejs/lib/bundling.ts

```diff
@@ -124,109 +123,52 @@ export interface ParcelOptions extends ParcelBaseOptions {
  */
 export class Bundling {
   /**
-   * Parcel bundled Lambda asset code
+   * esbuild bundled Lambda asset code
    */
-  public static parcel(options: ParcelOptions): lambda.AssetCode {
+  public static esbuild(options: EsBuildOptions): lambda.AssetCode {
     // Find project root
-    const projectRoot = options.projectRoot
-      ?? findUp(`.git${path.sep}`)
-      ?? findUp(LockFile.YARN)
-      ?? findUp(LockFile.NPM)
-      ?? findUp('package.json');
-    if (!projectRoot) {
-      throw new Error('Cannot find project root. Please specify it with `projectRoot`.');
-    }
-    const relativeEntryPath = path.relative(projectRoot, path.resolve(options.entry));
-
-    const packageJsonManager = new PackageJsonManager(path.dirname(options.entry));
-
-    // Collect external and install modules
-    let includeNodeModules: { [key: string]: boolean } | undefined;
-    let dependencies: { [key: string]: string } | undefined;
-    const externalModules = options.externalModules ?? ['aws-sdk'];
-    if (externalModules || options.nodeModules) {
-      const modules = [...externalModules, ...options.nodeModules ?? []];
-      includeNodeModules = {};
-      for (const mod of modules) {
-        includeNodeModules[mod] = false;
-      }
-      if (options.nodeModules) {
-        dependencies = packageJsonManager.getVersions(options.nodeModules);
-      }
-    }
-
-    let installer = Installer.NPM;
-    let lockFile: LockFile | undefined;
-    if (dependencies) {
-      // Use npm unless we have a yarn.lock.
-      if (fs.existsSync(path.join(projectRoot, LockFile.YARN))) {
-        installer = Installer.YARN;
-        lockFile = LockFile.YARN;
-      } else if (fs.existsSync(path.join(projectRoot, LockFile.NPM))) {
-        lockFile = LockFile.NPM;
-      }
-    }
+    const projectRoot = this.findProjectRoot(options.projectRoot);
 
-    // Configure target in package.json for Parcel
-    packageJsonManager.update({
-      targets: {
-        'cdk-lambda': {
-          context: 'node',
-          includeNodeModules: includeNodeModules ?? true,
-          sourceMap: options.sourceMaps ?? false,
-          minify: options.minify ?? false,
-          engines: {
-            node: `>= ${runtimeVersion(options.runtime)}`,
-          },
-        },
-      },
+    return lambda.Code.fromAsset(projectRoot, {
+      assetHashType: cdk.AssetHashType.OUTPUT,
+      bundling: new EsBuildBundler({
+        ...options,
+        externals: options.externalModules ?? ['aws-sdk'],
```

Comment:
I think we should not allow overwrite `['aws-sdk']`, it does not make much sense. in other words, it should be excluded anyway

Reply:
I am not sure about this. For my lambdas, I always package `aws-sdk` as I do not want to rely on the version the Lambda runtime provides.

Reply:
Agree with @hoegertn here, for critical production Lambdas users could want to opt-out.